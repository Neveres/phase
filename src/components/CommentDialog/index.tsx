export { default, default as CommentDialog } from './CommentDialog'
