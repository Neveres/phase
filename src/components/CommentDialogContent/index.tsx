export {
  default,
  default as CommentDialogContent,
} from './CommentDialogContent'
