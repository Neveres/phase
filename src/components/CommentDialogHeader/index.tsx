export { default, default as CommentDialogHeader } from './CommentDialogHeader'
