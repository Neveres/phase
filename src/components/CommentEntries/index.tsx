export { default, default as CommentEntries } from './CommentEntries'
