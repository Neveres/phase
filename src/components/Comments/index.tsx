export { default, default as Comments } from './Comments'
