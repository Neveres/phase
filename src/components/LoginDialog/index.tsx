export { default, default as LoginDialog } from './LoginDialog'
