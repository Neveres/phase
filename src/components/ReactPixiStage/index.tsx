export { default, default as ReactPixiStage } from './ReactPixiStage'
