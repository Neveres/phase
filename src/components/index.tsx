export { ReactPixiStage } from './ReactPixiStage'
export { CommentEntries } from './CommentEntries'
export { CommentDialog } from './CommentDialog'
export { AppHeader } from './AppHeader'
export { LoginDialog } from './LoginDialog'
