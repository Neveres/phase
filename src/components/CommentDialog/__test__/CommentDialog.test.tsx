import React from 'react'
import { create } from 'react-test-renderer'
import { render, screen, fireEvent, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { CommentDialog } from 'src/components'

jest.useFakeTimers().setSystemTime(new Date('2020-01-01'))
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (node: any) => node,
}))

describe('CommentDialog', () => {
  describe('render testing', () => {
    describe('in close status', () => {
      test('should render well while no comment existed', () => {
        const testRenderer = create(
          <CommentDialog
            open={false}
            closeDialog={jest.fn()}
            clearCommentGroupID={jest.fn()}
            groupActions={
              {
                create: jest.fn(),
                update: jest.fn(),
                delete: jest.fn(),
              } as any
            }
          />,
        )
        expect(testRenderer.toJSON()).toMatchSnapshot()
      })

      test('should render well while comments existed', () => {
        const testRenderer = create(
          <CommentDialog
            open={false}
            closeDialog={jest.fn()}
            clearCommentGroupID={jest.fn()}
            groupActions={
              {
                create: jest.fn(),
                update: jest.fn(),
                delete: jest.fn(),
              } as any
            }
            commentGroup={{
              uuid: 'uuid',
              comments: [
                {
                  name: 'name',
                  message: 'message',
                  postTime: 'postTime',
                },
              ],
              isResolved: false,
              coordinate: [100, 100],
            }}
          />,
        )
        expect(testRenderer.toJSON()).toMatchSnapshot()
      })
    })

    describe('in open status', () => {
      test('should render well while no comment existed', () => {
        const testRenderer = create(
          <CommentDialog
            open={true}
            closeDialog={jest.fn()}
            clearCommentGroupID={jest.fn()}
            groupActions={
              {
                create: jest.fn(),
                update: jest.fn(),
                delete: jest.fn(),
              } as any
            }
          />,
        )
        expect(testRenderer.toJSON()).toMatchSnapshot()
      })

      test('should render well while comments existed', () => {
        const testRenderer = create(
          <CommentDialog
            open={true}
            closeDialog={jest.fn()}
            clearCommentGroupID={jest.fn()}
            groupActions={
              {
                create: jest.fn(),
                update: jest.fn(),
                delete: jest.fn(),
              } as any
            }
            commentGroup={{
              uuid: 'uuid',
              comments: [
                {
                  name: 'name',
                  message: 'message',
                  postTime: 'postTime',
                },
              ],
              isResolved: false,
              coordinate: [100, 100],
            }}
          />,
        )
        expect(testRenderer.toJSON()).toMatchSnapshot()
      })
    })
  })

  describe('functional testing', () => {
    describe('while no comment existed', () => {
      test('adding new comment should work well ', async () => {
        const closeDialog = jest.fn()
        const clearCommentGroupID = jest.fn()
        const groupActions = {
          create: jest.fn(),
          update: jest.fn(),
          delete: jest.fn(),
        } as any

        render(
          <CommentDialog
            open={true}
            closeDialog={closeDialog}
            clearCommentGroupID={clearCommentGroupID}
            groupActions={groupActions}
          />,
        )

        fireEvent.change(screen.getByTestId('name-input'), {
          target: { value: 'name' },
        })

        fireEvent.change(screen.getByTestId('message-input'), {
          target: { value: 'message' },
        })

        fireEvent.click(screen.getByText('OK'))
        await waitFor(() => {
          expect(closeDialog).toBeCalled()
          expect(groupActions.create).toBeCalledWith({
            message: 'message',
            name: 'name',
            postTime: '1/1/2020, 8:00:00 AM',
          })
          expect(clearCommentGroupID).toBeCalled()
        })
      })

      test('clicking cancel should work', async () => {
        const closeDialog = jest.fn()
        const clearCommentGroupID = jest.fn()
        const groupActions = {
          create: jest.fn(),
          update: jest.fn(),
          delete: jest.fn(),
        } as any

        render(
          <CommentDialog
            open={true}
            closeDialog={closeDialog}
            clearCommentGroupID={clearCommentGroupID}
            groupActions={groupActions}
          />,
        )

        fireEvent.click(screen.getByText('Cancel'))
        await waitFor(() => {
          expect(closeDialog).toBeCalled()
          expect(clearCommentGroupID).toBeCalled()
        })
      })
    })

    describe('while comment existed', () => {
      test('adding new comment should work well', async () => {
        const closeDialog = jest.fn()
        const clearCommentGroupID = jest.fn()
        const groupActions = {
          create: jest.fn(),
          update: jest.fn(),
          delete: jest.fn(),
        } as any

        render(
          <CommentDialog
            open={true}
            closeDialog={closeDialog}
            clearCommentGroupID={clearCommentGroupID}
            groupActions={groupActions}
            commentGroup={{
              uuid: 'uuid',
              comments: [
                {
                  name: 'name',
                  message: 'message',
                  postTime: 'postTime',
                },
              ],
              isResolved: false,
              coordinate: [100, 100],
            }}
          />,
        )

        fireEvent.change(screen.getByTestId('name-input'), {
          target: { value: 'name' },
        })

        fireEvent.change(screen.getByTestId('message-input'), {
          target: { value: 'message' },
        })

        fireEvent.click(screen.getByText('OK'))
        await waitFor(() => {
          expect(closeDialog).toBeCalled()
          expect(groupActions.update).toBeCalledWith({
            isResolved: false,
            comment: {
              message: 'message',
              name: 'name',
              postTime: '1/1/2020, 8:00:00 AM',
            },
          })
          expect(clearCommentGroupID).toBeCalled()
        })
      })

      test('clicking cancel should work well', async () => {
        const closeDialog = jest.fn()
        const clearCommentGroupID = jest.fn()
        const groupActions = {
          create: jest.fn(),
          update: jest.fn(),
          delete: jest.fn(),
        } as any

        render(
          <CommentDialog
            open={true}
            closeDialog={closeDialog}
            clearCommentGroupID={clearCommentGroupID}
            groupActions={groupActions}
            commentGroup={{
              uuid: 'uuid',
              comments: [
                {
                  name: 'name',
                  message: 'message',
                  postTime: 'postTime',
                },
              ],
              isResolved: false,
              coordinate: [100, 100],
            }}
          />,
        )

        const checkbox = screen.getByTestId('resolved-switch')
        fireEvent.click(checkbox, {
          target: { checked: true },
        })
        expect(checkbox).toBeChecked()

        fireEvent.click(screen.getByText('Cancel'))
        await waitFor(() => {
          expect(closeDialog).toBeCalled()
          expect(clearCommentGroupID).toBeCalled()
        })
      })

      test('delete comment should work well', async () => {
        const closeDialog = jest.fn()
        const clearCommentGroupID = jest.fn()
        const groupActions = {
          create: jest.fn(),
          update: jest.fn(),
          delete: jest.fn(),
        } as any

        render(
          <CommentDialog
            open={true}
            closeDialog={closeDialog}
            clearCommentGroupID={clearCommentGroupID}
            groupActions={groupActions}
            commentGroup={{
              uuid: 'uuid',
              comments: [
                {
                  name: 'name',
                  message: 'message',
                  postTime: 'postTime',
                },
              ],
              isResolved: false,
              coordinate: [100, 100],
            }}
          />,
        )

        userEvent.hover(screen.getByText('Options'))
        await waitFor(() => {
          const deleteButton = screen.getByText('Delete')
          expect(deleteButton).toBeInTheDocument()
          fireEvent.click(deleteButton)
        })

        await waitFor(() => {
          expect(closeDialog).toBeCalled()
          expect(groupActions.delete).toBeCalled()
          expect(clearCommentGroupID).toBeCalled()
        })
      })
    })
  })
})
