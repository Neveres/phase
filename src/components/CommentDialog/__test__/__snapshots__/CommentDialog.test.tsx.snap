// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentDialog render testing in close status should render well while comments existed 1`] = `null`;

exports[`CommentDialog render testing in close status should render well while no comment existed 1`] = `null`;

exports[`CommentDialog render testing in open status should render well while comments existed 1`] = `
<div
  className="ant-modal-root"
>
  <div
    className="ant-modal-mask"
    style={
      Object {
        "zIndex": undefined,
      }
    }
  />
  <div
    className="ant-modal-wrap"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "display": null,
        "zIndex": undefined,
      }
    }
    tabIndex={-1}
  >
    <div
      aria-labelledby="test-id"
      aria-modal="true"
      className="ant-modal"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      role="dialog"
      style={
        Object {
          "width": 520,
        }
      }
    >
      <div
        aria-hidden="true"
        style={
          Object {
            "height": 0,
            "outline": "none",
            "overflow": "hidden",
            "width": 0,
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-modal-content"
      >
        <div
          className="ant-modal-header"
        >
          <div
            className="ant-modal-title"
            id="test-id"
          >
            <div
              className="css-rxdnkl-CommentDialog"
            >
              <div>
                <button
                  aria-checked={false}
                  className="ant-switch"
                  data-testid="resolved-switch"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="switch"
                  type="button"
                >
                  <div
                    className="ant-switch-handle"
                  />
                  <span
                    className="ant-switch-inner"
                  />
                </button>
                <span>
                  Resolved
                </span>
              </div>
              <div
                className="ant-dropdown-trigger"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                Options
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-modal-body"
        >
          <div
            className="css-8zdk6e-CommentDialog"
          >
            <div>
              <div
                className="comment-header"
              >
                <div
                  className="comment-name"
                >
                  name
                </div>
                <div
                  className="comment-post-time"
                >
                  postTime
                </div>
              </div>
              <div
                className="comment-message"
              >
                message
              </div>
            </div>
          </div>
          <input
            className="ant-input"
            data-testid="name-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Name"
            style={
              Object {
                "marginBottom": "5px",
              }
            }
            type="text"
            value=""
          />
          <input
            className="ant-input"
            data-testid="message-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Message"
            type="text"
            value=""
          />
        </div>
        <div
          className="ant-modal-footer"
        >
          <button
            className="ant-btn ant-btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
          <button
            className="ant-btn ant-btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              OK
            </span>
          </button>
        </div>
      </div>
      <div
        aria-hidden="true"
        style={
          Object {
            "height": 0,
            "outline": "none",
            "overflow": "hidden",
            "width": 0,
          }
        }
        tabIndex={0}
      />
    </div>
  </div>
</div>
`;

exports[`CommentDialog render testing in open status should render well while no comment existed 1`] = `
<div
  className="ant-modal-root"
>
  <div
    className="ant-modal-mask"
    style={
      Object {
        "zIndex": undefined,
      }
    }
  />
  <div
    className="ant-modal-wrap"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "display": null,
        "zIndex": undefined,
      }
    }
    tabIndex={-1}
  >
    <div
      aria-labelledby="test-id"
      aria-modal="true"
      className="ant-modal"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      role="dialog"
      style={
        Object {
          "width": 520,
        }
      }
    >
      <div
        aria-hidden="true"
        style={
          Object {
            "height": 0,
            "outline": "none",
            "overflow": "hidden",
            "width": 0,
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-modal-content"
      >
        <div
          className="ant-modal-header"
        >
          <div
            className="ant-modal-title"
            id="test-id"
          >
            Add comments
          </div>
        </div>
        <div
          className="ant-modal-body"
        >
          
          <input
            className="ant-input"
            data-testid="name-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Name"
            style={
              Object {
                "marginBottom": "5px",
              }
            }
            type="text"
            value=""
          />
          <input
            className="ant-input"
            data-testid="message-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Message"
            type="text"
            value=""
          />
        </div>
        <div
          className="ant-modal-footer"
        >
          <button
            className="ant-btn ant-btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
          <button
            className="ant-btn ant-btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              OK
            </span>
          </button>
        </div>
      </div>
      <div
        aria-hidden="true"
        style={
          Object {
            "height": 0,
            "outline": "none",
            "overflow": "hidden",
            "width": 0,
          }
        }
        tabIndex={0}
      />
    </div>
  </div>
</div>
`;
